### DISCUSSION TITLE
Need help with SQL query

### DISCUSSION THREAD

**Discussion by musiccollect** [filamentphp/filament#11994] [Help] [answered]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

I've tried every combination of getEloquentQuery, applySearchToTableQuery and modifyQueryUsing but getting nowhere. If I hardcode the value of $tableSearch, the table pulls the results from the database almost instantly but typing the value in the tableSearch box (as a user would do), takes ages to pull up the results. Same thing with or without DebugBar. I've tried this in both the resource page and the resource's list page. This is currently in the resource page:

public static function table(Table $table): Table
{
$tableSearch = request()->input('tableSearch');
$tableSearch = "The Dark Side Of The Moon";
return $table
->columns([
Tables\Columns\TextColumn::make('id')
->label('Release Track ID')
->hidden(),
Tables\Columns\TextColumn::make('title')
->sortable()
->searchable(isIndividual: false, isGlobal: true),
])
->searchable()
->paginated([10, 25, 50])
->defaultPaginationPageOption(10)
->searchOnBlur()
->modifyQueryUsing(fn (Builder $query) => $query->where('title', 'like', $tableSearch . '%'));
}

Any ideas ? 

Thanks, Greg

**Comment by leandrocfe** [comment, marked as answer]
You can inject `$request`
```php
->modifyQueryUsing(fn (Builder $query, Request $request) => $query->where('title', 'like', '%'.$request->tableSearch.'%'))
```

**Comment by musiccollect** [comment]
Thanks for the quick response. Using that code I get :

Target class [App\Filament\Resources\Request] does not exist.

**Reply by leandrocfe** [reply]
`use Illuminate\Http\Request;`

**Reply by musiccollect** [reply]
Thanks again. I added the use import but still same issue to begin with. I'm now looking at various options with Livewire on the frontend. Testing in that appears to be lightning-fast and could be the solution.

**Comment by flightphysical** [comment]
I don't have a solution yet, am trying to make sure I have imported the correct Builder Class, I am trying this one with intermittent results, not fully solved yet:

use Illuminate\Contracts\Database\Query\Builder;

anyone get this working yet?
